plotly_calplot/date_extractors.py

```diff
@@ -13,26 +13,16 @@ def get_date_coordinates(
 ) -> Tuple[Any, List[float], List[int]]:
     month_days = []
     for m in data[x].dt.month.unique():
-        month_days.append(data.loc[data[x].dt.month == m].max()[x].day)
+        month_days.append(data.loc[data[x].dt.month == m, x].max().day)
 
     month_positions = (np.cumsum(month_days) - 15) / 7
     weekdays_in_year = [i.weekday() for i in data[x]]
 
     # sometimes the last week of the current year conflicts with next year's january
-    # pandas will give those weeks the number 52 or 53, but this is bad news for this plot
-    # therefore we need a correction, for a more in-depth explanation check
+    # pandas uses ISO weeks, which will give those weeks the number 52 or 53, but this
+    # is bad news for this plot therefore we need a correction to use Gregorian weeks,
+    # for a more in-depth explanation check
     # https://stackoverflow.com/questions/44372048/python-pandas-timestamp-week-returns-52-for-first-day-of-year
-
-    weeknumber_of_dates = (
-        data[x].apply(lambda x: get_weeknumber_of_date(x)).values.tolist()
-    )
+    weeknumber_of_dates = data[x].dt.strftime("%W").astype(int).tolist()
```

Comment:
I'm happy to undo this if you prefer, but I think `.dt.strftime("%W").astype(int)` is a bit more concise (and maybe more performant, though likely insignificant for these small dfs).